appmap/_implementation/generation.py

```diff
@@ -114,5 +115,8 @@ def default(self, o):
 
 
 def dump(recording, metadata=None, indent=None):
+    """Generate a JSON representation of an appmap."""
+    if hasattr(recording, 'events'):
```

Comment:
Changing `appmap` to take `events` instead of `recording` makes sense to me. It seems to me that the corresponding change should be made here: change `recording` to `events`, and get rid of the `hasattr` check. I don't think we need to worry about the fact that this changes the interface.

Reply:
It's a public interface and I don't think there's a good reason to break it.

Reply:
I doubt there's anybody using it at this point. I think we should fix it before others start to use it.

Reply:
I also think it would make the interface worse, cf.
```python
rec = appmap.Recording()
with rec:
  ...
print(appmap.generation.dump(rec))
```
vs.
```python
print(appmap.generation.dump(rec.events))
```